queue.go

```diff
@@ -94,18 +119,23 @@ type Queue struct {
 }
 
 // NewQueue creates and returns a new Queue structure
-func NewQueue(url string, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+func NewQueue(name string, prefetchSize int, isConsumer, isExchange, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
```

Comment:
This name is misleading now because it can be a queue or an exchange. Clients will be using this module either as a queue or as an exchange so maybe two NewQueue() and NewExchange() methods that internally call one NewChannel(isConsumer) with this implementation would be better? This way the interface will be unambiguous